sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1086,7 +1086,7 @@ class Analyzer(override val catalogManager: CatalogManager)
         }
         view.copy(child = newChild)
       case p @ SubqueryAlias(_, view: View) =>
-        p.copy(child = resolveViews(view))
+        p.makeCopy(Array(p.identifier, resolveViews(view)))
```

Comment:
This is to copy the tags, but seems hacky since `copy` can be called to remove the tags in other places. Please check https://github.com/apache/spark/pull/31273#discussion_r561607183 if it makes sense to add `isView` to `SubqueryAlias`.